File: Best_Time_To_Buy_And_Sell_Stock/best_time_to_buy_and_sell_stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)

        fast = 1
        minPrice = prices[0]
        profit = 0
        while (fast < n):
            if prices[fast] < minPrice:
                minPrice = prices[fast]

            elif prices[fast] - minPrice > profit:
                profit = prices[fast] - minPrice
        
            fast += 1
        return profit

File: Best_Time_To_Buy_And_Sell_Stock/best_time_to_buy_and_sell_stock.c
int maxProfit(int* prices, int pricesSize) {

    int minPrice = prices[0];
    int maxProf = 0;

    for (int i = 0; i < pricesSize; i++) {
        if (prices[i] < minPrice) {
            minPrice = prices[i];
        } else if (prices[i] - minPrice > maxProf) {
            maxProf = prices[i] - minPrice;
        }

    }
    return maxProf;
}


